rotkehlchen/db/utils.py

```diff
@@ -372,3 +389,69 @@ def combine_asset_balances(balances: List[SingleDBAssetBalance]) -> List[SingleD
             new_balances = _append_or_combine(new_balances, balance)
 
     return new_balances
+
+
+def db_schema_sanity_check(
+        db_name: Literal['encrypted', 'global'],
+        cursor: 'DBCursor',
+) -> None:
+    """Ensures that database schema is not broken. Raises DBSchemaError if anything is bad."""
+    # Get mapping that contains expected database structure
+    tables_data_from_constant = MINIMIZED_ENCRYPTED_DB_SCHEMA if db_name == 'encrypted' else MINIMIZED_GLOBAL_DB_SCHEMA  # noqa: E501
```

Comment:
Why do this if? Just pass the mapping as an argument.

Or even better. I see you are passing a `DBCursor` while this is absolutely not needed here. In all places you use it you essentially create the cursor only for this function.

What you can do instead is a bit more advanced but would be better.

1. Go to the `DBConnection` class itself. At its initialization you actually know the type of the DB already. No need to pass it. Create a  pointer to the above schema, depending on the type at connection creation.
2. Either pass the `DBConnection` to this function, or make the sanity check a method of the DB connection instead. Probaly making it a method of the connection is cleaner.